src/containers/ProfilePage/ProfilePage.js

```diff
@@ -294,17 +296,20 @@ export const ProfilePageComponent = props => {
     return <NamedRedirect name="LandingPage" />;
   }
 
-  const isDataLoaded = isPreview
-    ? currentUser != null || userShowError != null
-    : user != null || userShowError != null;
   const isCurrentUser = currentUser?.id && currentUser?.id?.uuid === pathParams.id;
+  const isDataLoaded =
+    isPreview || (isCurrentUser || useCurrentUser)
+      ? currentUser != null || userShowError != null
+      : user != null || userShowError != null;
```

Comment:
This is not understandable.

The original setup tries to say that if page is rendered as a "preview" variant (e.g. user is in pending-approval state), then there must be a currentUser entity (it can't be null or undefined) and no errors when the call was made.

If the site is public, then user entity is the important data.

This now looks very strange and there's a hint of circular reasoning.

How about expanding this ternary to optional chaining setup?

```js
const isDataLoaded = isPreview
  ? currentUser != null || userShowError != null
  : hasNoViewingRightsOnPrivateMarketplace
  ? currentUser != null || userShowError != null
  : user != null || userShowError != null;
```

(and this could be moved below hasNoViewingRightsOnPrivateMarketplace)

Reply:
Updated!